core/runner/src/main/java/com/cognifide/aet/runner/processing/steps/CollectDispatcher.java

```diff
@@ -73,10 +73,17 @@ public void process() throws JMSException {
     Deque<MessageWithDestination> messagesQueue = Queues.newArrayDeque();
     LOGGER.info("Starting processing new Test Suite. CorrelationId: {} ", correlationId);
 
-    for (Test test : suite.get().getTests()) {
-      processUrlsAndGroupToPackages(messagesQueue, test);
+    for (MetadataRunDecorator metadataRunDecorator : runIndexWrapper.getUrls()) {
+      UrlRunWrapper urlRunWrapper = (UrlRunWrapper) metadataRunDecorator.getDecoratedRun();
+      final CollectorJobData data = new CollectorJobData(metadataRunDecorator.getCompany(),
+          metadataRunDecorator.getProject(), metadataRunDecorator.getName(), urlRunWrapper.getTestName(),
+          new ArrayList<>(Collections.singleton(urlRunWrapper.getObjectToRun())),
+          urlRunWrapper.getProxy(), urlRunWrapper.getPreferredBrowserId());
+      ObjectMessage message = session.createObjectMessage(data);
+      message.setJMSCorrelationID(correlationId);
+      messagesQueue.add(new MessageWithDestination(getQueueOut(), message, 1));
```

Comment:
You may refactor `MessageWithDestination` as well, to hold always only one message. But it could be done in a separate PR.